apps/managedidentity/managedidentity_test.go

```diff
@@ -274,3 +588,234 @@ func TestCreatingIMDSClient(t *testing.T) {
 		})
 	}
 }
+
+func TestAzureArcUserAssignedFailure(t *testing.T) {
+	tests := []struct {
+		name    string
+		id      ID
+		wantErr bool
+	}{
+		{
+			wantErr: true,
+			name:    "Client ID",
+			id:      UserAssignedClientID("test-client-id"),
+		},
+		{
+			wantErr: true,
+			name:    "Resource ID",
+			id:      UserAssignedResourceID("test-resource-id"),
+		},
+		{
+			wantErr: true,
+			name:    "Object ID",
+			id:      UserAssignedObjectID("test-object-id"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			unsetEnvVars(t)
+			setEnvVars(t, AzureArc)
+			client, err := New(tt.id)
+			if tt.wantErr {
+				if err == nil {
+					t.Fatal("client New() should return a error but did not.")
+				}
+
+				if err.Error() != "azure Arc doesn't support user assigned managed identities" {
+					t.Fatalf("expected error message 'azure Arc doesn't support user assigned managed identities', got %s", err.Error())
+				}
+				return
+			}
+			if err != nil {
+				t.Fatal(err)
+			}
+			if client.miType.value() != tt.id.value() {
+				t.Fatal("client New() did not assign a correct value to type.")
+			}
+		})
+	}
+}
+
+func TestValidateAzureArcEnvironment(t *testing.T) {
+	testCases := []struct {
+		name             string
+		identityEndpoint string
+		imdsEndpoint     string
+		platform         string
+		expectedResult   bool
+	}{
+		{
+			name:             "Both endpoints provided",
+			identityEndpoint: "endpoint",
+			imdsEndpoint:     "endpoint",
+			platform:         runtime.GOOS,
+			expectedResult:   true,
+		},
+		{
+			name:             "Only identityEndpoint provided",
+			identityEndpoint: "endpoint",
+			imdsEndpoint:     "",
+			platform:         runtime.GOOS,
+			expectedResult:   false,
+		},
+		{
+			name:             "Only imdsEndpoint provided",
+			identityEndpoint: "",
+			imdsEndpoint:     "endpoint",
+			platform:         runtime.GOOS,
+			expectedResult:   false,
+		},
+		{
+			name:             "No endpoints provided",
+			identityEndpoint: "",
+			imdsEndpoint:     "",
+			platform:         runtime.GOOS,
+			expectedResult:   false,
+		},
+		{
+			name:             "Platform not supported",
+			identityEndpoint: "",
+			imdsEndpoint:     "",
+			platform:         "darwin",
+			expectedResult:   false,
+		},
+		{
+			name:             "File does not exist",
+			identityEndpoint: "",
+			imdsEndpoint:     "",
+			platform:         runtime.GOOS,
+			expectedResult:   false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			setCustomAzureArcFilePath(t, fakeAzureArcFilePath)
+
+			result := isAzureArcEnvironment(tc.identityEndpoint, tc.imdsEndpoint, tc.platform)
```

Comment:
would be better to test this through public API i.e. `GetSource`--more durable, covers the same code--as in `TestAzureArcReturnsWhenHimdsFound`. Actually, looks like this test should merge with that one

Reply:
Deleted this test